### PR TITLE
K8SPG-390 implement dynamic test execution

### PR DESCRIPTION
**Pull request by hors**
[K8SPG-390](https://jira.percona.com/browse/K8SPG-390) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
*Short explanation of the problem.*

**Cause:**
*Short explanation of the root cause of the issue if applicable.*

**Solution:**
*Short explanation of the solution we are providing with this PR.*

**CHECKLIST**
---
**Jira**
- [ ] Is the Jira ticket created and referenced properly?
- [ ] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [ ] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/monitoring.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-473/30d3aa1a8/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/473/commits/30d3aa1a8648fe51aeae6f0cc5e475d9b4c0850b
image: `perconalab/percona-postgresql-operator:PR-473-30d3aa1a8`

File: Jenkinsfile [hunk 1]
```diff
@@ -80,93 +103,134 @@ void pushLogFile(String FILE_NAME) {
     }
 }
 
-void pushArtifactFile(String FILE_NAME) {
-    echo "Push $FILE_NAME file to S3!"
+void popArtifactFile(String FILE_NAME) {
+    echo "Try to get $FILE_NAME file from S3!"
```
[the remark this answers is not in the document]
Comment (hors) [reply Jenkinsfile]: fixed

File: Jenkinsfile [hunk 2]
```diff
@@ -80,93 +103,123 @@ void pushLogFile(String FILE_NAME) {
...
-    for ( test in testsReportMap ) {
-        TestsReport = TestsReport + "\r\n| ${test.key} | ${test.value} |"
+    def wholeTestAmount=tests.size()
+    def startedTestAmount = 0
+
+    for (int i=0; i<tests.size(); i++) {
+        def testName = tests[i]["name"]
+        def testResult = tests[i]["result"]
+        def testTime = tests[i]["time"]
+        def testUrl = "${testUrlPrefix}/${env.GIT_BRANCH}/${env.GIT_SHORT_COMMIT}/${testName}.log"
```
Comment (nmarukovich) [inline Jenkinsfile:149]: Why we use testUrlPrefix var if it only needs us in one place?

File: Jenkinsfile [hunk 3]
```diff
@@ -26,20 +27,29 @@ void CreateCluster(String CLUSTER_SUFFIX, String SUBNETWORK = CLUSTER_SUFFIX) {
...
+                kubectl delete pods --all -n \$namespace --force --grace-period=0 || true
+            done
+            kubectl get svc --all-namespaces || true
             gcloud container clusters delete --zone $GKERegion $CLUSTER_NAME-${CLUSTER_SUFFIX}
         """
    }
 }
 
-void DeleteOldClusters(String FILTER) {
+void deleteOldClusters(String FILTER) {
```
Comment (nmarukovich) [inline Jenkinsfile:52]: If it is not big deal, could you please add CW option? (We can skip for now)
```
void isRunTestsInClusterWide() {
    if ("${params.CLUSTER_WIDE}" == "YES") {
        env.OPERATOR_NS = 'pg-operator'
    }
}
```

Comment (hors) [reply Jenkinsfile:52]: it is Jenkinsfile we run all tests in CW by default

File: Jenkinsfile (hunk 2 above)
Comment (hors) [reply Jenkinsfile:149]: I do not want to have long string + in all our operators we have it. I do not want to create a difference.

**Review by nmarukovich — APPROVED** [review]

**Pull request merged**